CONTRIBUTING.md

```diff
@@ -0,0 +1,39 @@
+# Contributing
+**Table of Contents**
+- [Getting Started](#getting-started) (1st draft)
+- [Philosophy](#philosophy) (TBD)
+- [Reporting an Issue](#reporting-an-issue) (TBD)
+- [Pull Requests](#pull-requests) (1st draft)
+	-  _Should this be renamed to "fixing an issue"?_
+	-  Link to GitHub guide for PRs
+	-  What issues are good to start with? (Eg. `feature`, `bug`, etc. )
+- [Setup and Run](#setup-and-run) (TBD)
+## Getting Started
+Hi, and welcome to highlight.js. We're a code-highlighting engine etc., but you probably know that. You want to contribute, and this document is the your guide.
+
+There are many ways to contribute to highlight.js. Feel free to work on an open issue. You can also help by writing [a plugin](https://highlightjs.readthedocs.io/en/latest/plugin-api.html) or creating [a theme](https://highlightjs.readthedocs.io/en/latest/style-guide.html). In terms of getting help, GitHub issues are the primary method of communication. We're exploring the idea of creating a Gitter chat room.
+
+## Philosophy
+> A short section on our overall philosophy somewhere would be great also (I'm happy to help there):
+> - keep the core parser simple (focus on the 80%)
+> - encourage and support a plug-in culture (let others tackle the last 20%)
+> - we don't show line numbers on purpose
+> - we're more than a keyword highlighter
+> - we're not a full language parser
+> - auto-detect is not powered by magic unicorn dust
+> 
+> ([Josh Goebel](https://github.com/highlightjs/highlight.js/issues/2753#issuecomment-709415929))
+
+## Reporting an Issue
+
+## Pull Requests
+When opening a PR, follow the template.
```

Comment:
This should probably be two lists... "before you open a PR" and then then "when" list...

"Open an issue first, so we can explore and discuss it"

This is worth a paragraph.  I can flesh it out if you just get something down.